libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerTests.java

```diff
@@ -153,12 +154,16 @@ public void qnaMakerTraceActivity() {
                     .assertReply("echo:bar")
                 .startTest().join();
 
+            ObjectMapper objectMapper = new ObjectMapper()
+                .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+                .findAndRegisterModules();
+
             // Validate Trace Activity created
             PagedResult<Activity> pagedResult = transcriptStore.getTranscriptActivities("test", conversationId[0]).join();
             Assert.assertEquals(7, pagedResult.getItems().size());
             Assert.assertEquals("how do I clean the stove?", pagedResult.getItems().get(0).getText());
             Assert.assertTrue(pagedResult.getItems().get(1).isType(ActivityTypes.TRACE));
-            QnAMakerTraceInfo traceInfo = (QnAMakerTraceInfo) pagedResult.getItems().get(1).getValue();
+            QnAMakerTraceInfo traceInfo = objectMapper.readValue(objectMapper.writeValueAsString(pagedResult.getItems().get(1).getValue()), QnAMakerTraceInfo.class);
```

Comment:
Do you know why the test failed? It was because our changes? I don't think so.. so why?

Reply:
I really don't know. I debugged the test and everything was ok, it was not these last changes that caused this error, since in the base branch it was too. It was throwing an exception: `java.util.LinkedHashMap cannot be cast to com.microsoft.bot.ai.qna.models.QnAMakerTraceInfo`, since this `pagedResult.getItems().get(1).getValue()` was returning a LinkedHashMap